### PR TITLE
Remove `lfortran_intrinsic_math.f90` file and implement `system_clock`

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/4593

### CODE REVIEW

File: src/libasr/runtime/lfortran_intrinsics.c [hunk 1]
```diff
@@ -2420,76 +2420,78 @@ LFORTRAN_API float _lfortran_s_cpu_time() {
...
-LFORTRAN_API void _lfortran_i32sys_clock(
-        int32_t *count, int32_t *rate, int32_t *max) {
-#if defined(_WIN32)
-        *count = - INT_MAX;
-        *rate = 0;
-        *max = 0;
-#else
+LFORTRAN_API int32_t _lfortran_i32sys_clock_count() {
     struct timespec ts;
     if(clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
```
Comment (certik) [inline src/libasr/runtime/lfortran_intrinsics.c:2425]: This line was not compiled on Windows, but now it is, so it gives an error:
```
2024-10-02T09:41:07.2901743Z D:\a\lfortran\lfortran\lfortran-0.40.0-288-g714d92374\src\libasr\runtime\lfortran_intrinsics.c(2425): error C2065: 'CLOCK_MONOTONIC': undeclared identifier
2024-10-02T09:41:07.2906412Z D:\a\lfortran\lfortran\lfortran-0.40.0-288-g714d92374\src\libasr\runtime\lfortran_intrinsics.c(2434): error C2065: 'CLOCK_MONOTONIC': undeclared identifier
```

**Review by certik — APPROVED** [review]
Excellent, great job!

Two issues:
* the runtime library must be fixed to compile on Windows
* does `cpu_time` still work? If so, then this can be merged.

File: src/libasr/runtime/lfortran_intrinsics.c (hunk 1 above)
Comment (HarshitaKalani) [reply src/libasr/runtime/lfortran_intrinsics.c:2428]: Got it.

**Comment by HarshitaKalani** [comment]
Yep, I think this is ready now.

**Review by certik — APPROVED** [review]

**Pull request merged**